tortoise/backends/base/schema_generator.py

```diff
@@ -179,23 +179,26 @@ def _get_table_sql(self, model: "Type[Model]", safe: bool = True) -> dict:
                     if reference.description
                     else ""
                 )
+
+                field_type = reference.to_field.get_for_dialect(self.DIALECT, "SQL_TYPE")
+                field_name = reference.to_field.source_field
+                if not field_name:
+                    field_name = reference.to_field.model_field_name
+
                 field_creation_string = self._create_string(
                     db_field=db_field,
-                    field_type=field_object.get_for_dialect(self.DIALECT, "SQL_TYPE"),
+                    field_type=field_type,
```

Comment:
Oh dear, our field type depends on the DB backend...
We would want the field object so that we can get the SQL_TYPE lazily (as a model shouldn't be tightly bound to a DB type)
I didn't think of this?
Maybe we should also store the `to_field_instance`? :thinking: